v2/internal/util/mysql/privilege.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+)
+
+type SQLPrivilegeDelta struct {
+	AddedPrivileges   set.Set[string]
+	DeletedPrivileges set.Set[string]
+}
+
+const sqlAll = "ALL" // TODO: We don't actually have to special case this because it's not even supported in FlexibleServer...
```

Comment:
So do we need this at all?

Reply:
We don't really, unless we end up needing to expand support to include mySQL standalone server in which case having this would be helpful. I'm not sure we know _for sure_ we're not going to do that (we support it in ASOv1). The cost of me including this was low because it was already written for ASOv1 and I basically just adopted that code.

Happy to remove it if you think the cost is higher than it's worth. It is covered in the tests though so confidence that it does the right thing is pretty high.

Reply:
Basically this is "the library does more than we need, but we might need it some day and it's tested so is it worth removing?"

Reply:
Fair enough; leave it in.